fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergAwsClientFactory.java

```diff
@@ -151,8 +151,9 @@ public S3Client s3() {
             s3ClientBuilder.region(Region.of(s3Region));
         }
 
-        if (!s3Endpoint.isEmpty()) {
-            s3ClientBuilder.endpointOverride(URI.create(s3Endpoint));
+        URI uri;
+        if (!s3Endpoint.isEmpty() && (uri = URI.create(s3Endpoint)).getScheme() != null) {
```

Comment:
add comments better

Reply:
done

Reply:
if it is a valid URI, just lack of scheme, can we assume it is `HTTPS`?

e.g. if the override endpoint provided is `abcbucket.s3.us-east.amazonaws.com`, we will safely assume it is `https://abcbucket.s3.us-east.amazonaws.com`?

make the code robust while keep easy to use.

Reply:
It is not necessary, `endpointOverride` can be any valid S3 endpoint, including endpoints that use different protocols. Typically, AWS S3 services use the HTTPS protocol for secure data transmission, so in most cases, endpointOverride should also use the HTTPS protocol.

But if we can assume it is HTTPS, then in the `software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.endpointOverride` method, it should be done in this way that does not just check if the scheme is missing and just throw a NPE exception.

Reply:
This is just a java implementation of the sdk behavior. We can tune it to make it easy to use if necessary.

Ignoring the endpoint is better than NPE. But does it good though?

With this patch,  the endpoint override is ignored it in silence, user will not know the config item is ignored until something is not working. throwing ddlexception would be better than just ignoring.

As for using https if the scheme empty, it is the default behavior of cpp version of aws sdk, check the following code. I don't think it is not something we should not do, it is just how we define the behavior in such case.

https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/client/ClientConfiguration.h#L107
```
             * Http scheme to use. E.g. Http or Https. Default HTTPS
             */
            Aws::Http::Scheme scheme;

```

https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/endpoint/BuiltInParameters.cpp#L63
```
        if (!config.endpointOverride.empty()) {
            OverrideEndpoint(config.endpointOverride, config.scheme);
```

https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/endpoint/BuiltInParameters.cpp#L15
```
    void BuiltInParameters::OverrideEndpoint(const Aws::String& endpoint, const Aws::Http::Scheme& scheme)
    {
        static const Aws::String SDK_ENDPOINT = "Endpoint";

        if (endpoint.compare(0, 7, "http://") == 0 || endpoint.compare(0, 8, "https://") == 0)
        {
            SetStringParameter(SDK_ENDPOINT, endpoint);
        }
        else
        {
            SetStringParameter(SDK_ENDPOINT, Aws::String(Aws::Http::SchemeMapper::ToString(scheme)) + "://" + endpoint);
        }
    }

```

Reply:
done.
Got it! This is more robust.
Thanks for comments!